systemtest/src/test/java/io/strimzi/systemtest/dynamicconfiguration/DynamicConfigurationIsolatedST.java

```diff
@@ -0,0 +1,292 @@
+
+        updateAndVerifyDynConf("true");
+
+        LOGGER.info("Verify values after update");
+        kafkaConfiguration = kubeClient().getConfigMap(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).getData().get("server.config");
+        assertThat(kafkaConfiguration, containsString("offsets.topic.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("transaction.state.log.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("log.message.format.version=2.5"));
+        assertThat(kafkaConfiguration, containsString("unclean.leader.election.enable=true"));
+
+        InlineLogging il = new InlineLoggingBuilder().withLoggers(Collections.singletonMap("kafka.logger.level", "INFO")).build();
+
+        Map<String, String> kafkaPodsSnapshot = StatefulSetUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Updating logging of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaClusterSpec kafkaClusterSpec = k.getSpec().getKafka();
+            kafkaClusterSpec.setLogging(il);
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), KAFKA_REPLICAS, kafkaPodsSnapshot);
+    }
+
+    @Tag(NODEPORT_SUPPORTED)
+    @Test
+    void testDynamicConfigurationWithExternalListeners() {
+        KafkaResource.kafkaPersistent(CLUSTER_NAME, KAFKA_REPLICAS, 1)
+            .editSpec()
+            .editKafka()
+                .withNewListeners()
+                    .withNewKafkaListenerExternalNodePort()
+                        .withTls(false)
+                    .endKafkaListenerExternalNodePort()
+                    .withNewPlain()
+                    .endPlain()
+                .endListeners()
+                .withConfig(kafkaConfig)
+            .endKafka()
+            .endSpec()
+            .done();
+
+        updateAndVerifyDynConf("true");
+
+        // Edit listeners - this should cause RU (because of new crts)
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME));
+        LOGGER.info("Updating listeners of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaClusterSpec kafkaClusterSpec = k.getSpec().getKafka();
+            KafkaListeners kl = new KafkaListenersBuilder()
+                    .withNewKafkaListenerExternalNodePort()
+                    .endKafkaListenerExternalNodePort()
+                    .withNewPlain()
+                    .endPlain()
+                    .build();
+            kafkaClusterSpec.setListeners(kl);
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), KAFKA_REPLICAS, kafkaPods);
+        assertThat(StatefulSetUtils.ssHasRolled(kafkaStatefulSetName(CLUSTER_NAME), kafkaPods), is(true));
+
+        updateAndVerifyDynConf("false");
+        updateAndVerifyDynConf("true");
```

Comment:
Why are you calling both like this?

Reply:
This is just refactored tests, which just update the Dyn. configuration of the broker. The point is to make as many changes as possible. If I understand correctly from @stanlyDoge.

Reply:
Yeah, as I understand the test this two lines set the same property (`unclean.leader.election.enable`) to false and then to true. We are checking at the end of the test, whether kafka pods rolled.
I would consider changing another property.